graph/shape/shape.go

```diff
@@ -1395,3 +1395,28 @@ func (s Save) Optimize(r Optimizer) (Shape, bool) {
 	}
 	return s, opt
 }
+
+type Values struct {
+	Sub   []quad.Value
+	Pred  []quad.Value
+	Obj   []quad.Value
+	Label []quad.Value
+	From  Shape
+}
+
+func (v Values) BuildIterator(qs graph.QuadStore) graph.Iterator {
```

Comment:
It can be a function returning a shape:
```golang
func QuadsWithValues(sub, pred, obj, label []quad.Value) Shape {
```
And it will be a user's responsibility to call `BuildIterator` or use the shape in a more complex query.

Reply:
I'm not sure how to do it

Reply:
I mean, the `Quads` type is a shape, so I only propose to remove the new `Values` type, make a function instead of a method (no receiver) and do not call `BuildIterator` at the end, returning `q` directly. Note the signature I proposed though. We want to return `Shape` in general, not `Quads` specifically, so we don't lock ourself down with a specific implementation.

Reply:
Sounds correct. I'm just not fully sure how to implement it.

Reply:
No worries, let me hijack the PR then. Will push the changes here, or in case of issues will close the PR and merge directly to master (will preserve your authorship of course).